src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -664,4 +677,31 @@ mod tests {
             Datum::Time(Time::parse_utc_datetime("0000-00-00 00:00:00", 6).unwrap()),
         );
     }
+
+    #[test]
+    fn test_week_mode() {
+        let cases = vec![
+            ("2000-01-01", 0, 0i64),
+            ("2008-02-20", 0, 7i64),
+            ("2008-02-20 00:00:00", 1, 8i64),
+            ("2008-12-31 00:00:00", 1, 53i64),
```

Comment:
Can you add some non-edge tests? Like non-leap years?

Reply:
Sure,  I will add more test cases.

Reply:
👍